content/docs/warehouses-supported/bigquery.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Google BigQuery
+---
+
+# Authentification
+
+Dataform will connect to BigQuery using a service account. You’ll need to create a service account from your Google Cloud Console and assign it permissions to access BigQuery.
+
+1. Follow <a target="_blank" rel="noopener" href="https://cloud.google.com/iam/docs/creating-managing-service-accounts#creating_a_service_account">these instructions</a> to create a new service account in Google Cloud Console.
+2. Grant the new account the `BigQuery Admin` role. (Admin access is required by Dataform so that it can create queries and list tables.) Read
+   <a target="_blank" rel="noopener" href="https://cloud.google.com/iam/docs/granting-roles-to-service-accounts#granting_access_to_a_service_account_for_a_resource">this</a> if you need help.
+3. Create a key for your new service account (in JSON format). You will upload this file to Dataform. Read
+   <a target="_blank" rel="noopener" href="https://cloud.google.com/iam/docs/creating-managing-service-account-keys#creating_service_account_keys">this</a> if you need help.
+
+BigQuery specific options can be applied to tables using the `bigquery` configuration parameter.
+
+# Configuration options
+
+## Setting table partitions
+
+BigQuery supports <a target="_blank" rel="noopener" href="https://cloud.google.com/bigquery/docs/partitioned-tables">partitioned tables</a>.
+These can be useful when you have data spread across many different dates but usually query the table on only a small range of dates.
+In these circumstances, partitioning will increase query performance and reduce cost.
+
+BigQuery partitions can be configured in Dataform using the `partitionBy` option:
+
+```js
+config {
+  type: "table",
+  bigquery: {
+    partitionBy: "DATE(ts)"
+  }
+}
+SELECT CURRENT_TIMESTAMP() AS ts
+```
+
+This query compiles to the following statement which takes advantage of BigQuery's DDL to configure partitioning:
+
+```js
+CREATE OR REPLACE TABLE dataform.example
+PARTITION BY DATE(ts)
+AS (SELECT CURRENT_TIMESTAMP() AS ts)
+```
+
+If desired, partitions can be clustered by using the `clusterBy` option, for example:
+
+```js
+config {
+  type: "table",
+  bigquery: {
+    partitionBy: "DATE(ts)",
+    clusterBy: ["name", "revenue"]
+  }
+}
+SELECT CURRENT_TIMESTAMP() as ts, name, revenue
+```
+
+## Configuring access to Google Sheets
+
+In order to be able to query Google Sheets tables via BigQuery, you'll need to share the sheet with the service account that is used by Dataform.
+
+- Find the email address of the service account you connected Dataform with by looking in the `.df-credentials.json` file locally, or by finding the account from the [Google Cloud IAM service accounts console](https://console.cloud.google.com/iam-admin/serviceaccounts).
```

Comment:
```suggestion
- Find the email address of the service account through which you connected to Dataform. You can find this on the [Google Cloud IAM service accounts console page](https://console.cloud.google.com/iam-admin/serviceaccounts). If you're developing your Dataform project locally (as opposed to using Dataform Web), you can find the service accounts email in the `.df-credentials.json` file.
```

Reordered (most of our users are dataform web, so put that first), and clarified that the df-credentials file bit isn't relevant if you're using dataform web